hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -104,13 +101,16 @@
    */
   private long contentRangeStart;
 
+  private MultipartDownloader multipartDownloader;
+
   public S3AInputStream(S3ObjectAttributes s3Attributes,
-      long contentLength,
-      AmazonS3 client,
-      FileSystem.Statistics stats,
-      S3AInstrumentation instrumentation,
-      long readahead,
-      S3AInputPolicy inputPolicy) {
+                        long contentLength,
+                        AmazonS3 client,
+                        FileSystem.Statistics stats,
+                        S3AInstrumentation instrumentation,
+                        long readahead,
+                        S3AInputPolicy inputPolicy, MultipartDownloader multipartDownloader) {
+    this.multipartDownloader = multipartDownloader;
```

Comment:
can we move this to the bottom of the constructor to match the ordering of the params?